packages/components/inputs/SelectInput.tsx

```diff
@@ -37,34 +45,84 @@ type Props = {
   data: SelectInputData[];
   placeHolder?: string;
   selectedData: SelectInputData;
-  setData: (data: SelectInputData) => void;
+  selectData: (data: SelectInputData) => void;
```

Comment:
```suggestion
  onSelect: (data: SelectInputData<T>) => void;
```

Reply:
also it would better to use a generic for the data type https://www.typescriptlang.org/docs/handbook/2/generics.html